pylint/checkers/refactoring.py

```diff
@@ -261,13 +265,30 @@ def _check_redefined_argument_from_local(self, name_node):
                                  args=(name_node.name, ))
 
     @utils.check_messages('redefined-argument-from-local',
-                          'too-many-nested-blocks')
+                          'too-many-nested-blocks',
+                          'mutable-sequence-modified-in-loop')
     def visit_for(self, node):
         self._check_nested_blocks(node)
 
         for name in node.target.nodes_of_class(astroid.AssignName):
             self._check_redefined_argument_from_local(name)
 
+        inferrend = utils.safe_infer(node.iter)
+        if (isinstance(node.iter, astroid.Name) and
```

Comment:
What do you think about inverting condition and doing early-return? It's one less indentation level.
What do you think about extracting a static method? Are these checks can be named in meaningful name?